### PR TITLE
Refactor Pairwise term to precompute summation bounds

### PR DESCRIPTION
**Pull request by niklasschmitz**
This PR follows up on #656 now analogously on the Pairwise term, replacing the shell_indices & while loop by a dense for-loop over space. Later, this can also be now easily modified further into mapreduce-like patterns for GPU or AD.

### CODE REVIEW

File: src/structure.jl [hunk 1]
```diff
@@ -42,3 +42,15 @@ function diameter(lattice::AbstractMatrix)
     end
     diam
 end
+
+"""
+Estimate integer bounds for dense space loops from a given inequality ||Mx|| ≤ δ.
+For 1D and 2D systems the limit will be zero in the auxiliary dimensions.
+"""
+function estimate_integer_lattice_bounds(M, δ, shift=zeros(3))
```
Comment (antoine-levitt) [inline src/structure.jl:50]: Can you possibly also use it in fft.jl? See bounding_rectangle.jl

**Review by antoine-levitt — COMMENTED** [review]
Looks good! @epolack ?

File: src/structure.jl (hunk 1 above)
Comment (epolack) [reply src/structure.jl:50]: Will look into that.

**Comment by epolack** [comment]
Had to modify it a bit to not get NaN, otherwise good. I have rebased the complex version on this change.

**Pull request merged**